src/eth_provider/provider.rs

```diff
@@ -721,12 +720,40 @@ where
         block_id: impl Into<Option<BlockId>>,
     ) -> EthProviderResult<starknet::core::types::BlockId> {
         match block_id.into() {
-            Some(BlockId::Hash(hash)) => {
-                Ok(starknet::core::types::BlockId::Number(self.hash_into_block_number(hash).await?.to::<u64>()))
-            }
-            Some(id) => {
-                let eth_block_id = EthBlockId::new(id);
-                Ok(eth_block_id.try_into()?)
+            Some(BlockId::Hash(hash)) => Ok(EthBlockId::new(BlockId::Hash(hash)).try_into()?),
```

Comment:
Remove hash into block number because if blockhash is 0x0 -> block number will be out of range for SN block id